### ISSUE TITLE
Question about sRGB-XYZ pair generation

### ISSUE THREAD

**Issue by DY112** [mahmoudnafifi/CIE_XYZ_NET#9]
Hi, @mahmoudnafifi.
I have a question about sRGB-XYZ image pair generation.
I am currently generating my own stylized sRGB - XYZ image dataset, using NUS-8 dataset.
The stylized sRGB images are generated using Adobe lightroom classic, using RAW files.
And I generated XYZ images with DCRAW.
[using "dcraw -r {1/wb_kernel[0]} 1.0 {1/wb_kernel[2]} 1.0 -o 5 -g 1 1 -6 -T {raw_file}"]

However, I noticed that the XYZ and stylized sRGB images have different sizes.
I think the lens correction matrix might be the reason, but I'm not sure why.
I'd like the software to generate both images to be the same, but I'm stuck between lightroom for generating stylized sRGB and DCRAW or rawpy for generating XYZ images.
Can you think of anything that might be causing the size of the two images to mismatch?

![stylized_sRGB](https://github.com/mahmoudnafifi/CIE_XYZ_NET/assets/24367643/a2286e63-2485-459a-8f4e-9272a53e7ebd)

The size of stylized sRGB is 5616x3744.

![XYZ](https://github.com/mahmoudnafifi/CIE_XYZ_NET/assets/24367643/9053a2d9-7d0a-4acb-90a9-dfcc0a9e760a)

The size of XYZ is 5640x3752.

**Comment by mahmoudnafifi** [comment]
Hello, the different sizes you got occur due to cropping applied during image processing in the camera pipeline. This resizing can happen once at the photofinishing stage or as a series of cropping steps, with each step applied at an intermediate stage in the pipeline. Unfortunately there are no standardized practices in this regard. So it can be challenging to track these changes, especially when using a black box software like lightroom. Also not sure if dcraw applies some cropping even in early stages or not.

In our paper we generated the xyz data ourselves using this [simple ISP](https://github.com/AbdoKamel/simple-camera-pipeline). Then, we rendered the sRGB on adobe camera raw in photoshop. I am unsure if this process mirrors the rendering in lightroom or not. With said, I vaguely recall encountering some misalignments in some images/cameras and removed them from the dataset, though I can't be entirely certain -- it was done ~4 years ago.
